### PR TITLE
[filesystems]Support Tencent COSN (#842)

### PR DESCRIPTION
**Pull request by daixiaoyu**
*[filesystems]*

### Purpose

Support Tencent COSN

### Documentation

*(Does this change introduce a new feature)*
change paimon/docs/content/filesystems/overview.md
add       paimon/docs/content/filesystems/cosn.md

### CODE REVIEW

**Comment by SteNicholas** [comment]
@daixiaoyu, could you add the test cases for loader of Tencent COSN support?

**Comment by daixiaoyu** [comment]
@SteNicholas Okay, I'll try.

File: paimon-filesystems/paimon-cosn/src/main/java/org/apache/paimon/cosn/COSNLoader.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    private static final String COSN_CLASS = "org.apache.paimon.cosn.COSNFileIO";
+
+    // Singleton lazy initialization
+
+    private static PluginLoader loader;
+
+    private static synchronized PluginLoader getLoader() {
+        if (loader == null) {
+            // Avoid NoClassDefFoundError without cause by exception
+            loader = new PluginLoader(COSN_JAR);
```
Comment (SteNicholas) [inline paimon-filesystems/paimon-cosn/src/main/java/org/apache/paimon/cosn/COSNLoader.java:42]: Why is `COSN_JAR` fixed? Is it necessary to make the name to the final `paimon-plugin-cosn.jar`? Other jar name could not?

Comment (daixiaoyu) [reply paimon-filesystems/paimon-cosn/src/main/java/org/apache/paimon/cosn/COSNLoader.java:42]: Here I refer to the implementation of OSS, mainly to the code style can be unified

File: paimon-filesystems/paimon-cosn-impl/pom.xml [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+                        <configuration>
+                            <artifactSet>
+                                <includes>
+                                    <include>*:*</include>
+                                </includes>
+                            </artifactSet>
+                            <relocations>
+                                <relocation>
+                                    <pattern>org.apache.hadoop</pattern>
+                                    <shadedPattern>org.apache.flink.fs.shaded.hadoop3.org.apache.hadoop</shadedPattern>
```
Comment (JingsongLi) [inline paimon-filesystems/paimon-cosn-impl/pom.xml:105]: org.apache.paimon.shaded

**Review by JingsongLi — COMMENTED** [review]
Thanks @daixiaoyu for the contribution. Left one comment.

Can you explain what is COSN? I only know COS.

I know there is S3 client for COS.
So what is difference between S3 client and COSN client?

**Comment by SteNicholas** [comment]
@daixiaoyu, any comments or updates for above review comments?

**Comment by daixiaoyu** [comment]
This can be close , Tencent COS  can use  S3 FileSystem

**Pull request closed without merging**